RELEASE/scripts/autoscend/auto_equipment.ash

```diff
@@ -179,6 +179,266 @@ boolean tryAddItemToMaximize(slot s, item it)
 	return true;
 }
 
+item[slot] speculatedMaximizerEquipment(string statement)
+{
+	//make maximizer simulate with the given statement then return the list of equipment it has chosen
+	item [slot] res;
+	boolean weaponPicked;
+	boolean offhandPicked;
+	foreach i,entry in maximize(statement,0,0,true,true)	//can't use autoMaximize "Aggregate reference expected"
+	{
+		if(i>15)
+		{
+			//there should not be more than 9 or 10 equipment slots and equipment entries come first. so equipment list is done
+			break;
+		}
+		item maximizerItem = entry.item;
+		if(maximizerItem == $item[none]) continue;
+		slot maximizerItemSlot = maximizerItem.to_slot();
+		if(maximizerItemSlot == $slot[none]) continue;
+		slot overrideSlot;
+		if(maximizerItemSlot == $slot[weapon])
+		{
+			if(weaponPicked && offhandPicked)
+			{
+				//this must be familiar weapon
+				overrideSlot = $slot[familiar];
+			}
+			else if(weaponPicked)
+			{
+				//this must be offhand weapon
+				overrideSlot = $slot[off-hand];
+				offhandPicked = true;
+			}
+			else
+			{
+				weaponPicked = true;
+				if(weapon_hands(maximizerItem) > 1)	offhandPicked = true;
+			}
+		}
+		else if(maximizerItemSlot == $slot[off-hand])
+		{
+			if(offhandPicked)
+			{
+				//this must be familiar offhand
+				overrideSlot = $slot[familiar];
+			}
+			else
+			{
+				offhandPicked = true;
+			}
+		}
+		else if(maximizerItemSlot == $slot[acc1] && res[$slot[acc1]] != $item[none])
+		{
+			//accessory to slot always returns acc1 and has to be switched if more than one, go from 1 to 3 because that is the equip order the maximizer will use
+			if(res[$slot[acc2]] != $item[none])
+			{
+				overrideSlot = $slot[acc3];
+			}
+			else
+			{
+				overrideSlot = $slot[acc2];
+			}
+		}
+		if(overrideSlot != $slot[none])
+		{
+			maximizerItemSlot = overrideSlot;
+		}
+		if(res[maximizerItemSlot] != $item[none])
+		{
+			auto_log_debug("Duplicate entry skipped for slot " + maximizerItemSlot.to_string() + " in speculatedMaximizerEquipment, something must be wrong", "gold");
+			continue;
+		}
+		res[maximizerItemSlot] = maximizerItem;
+	}
+	return res;
+}
+
+void equipStatgainIncreasers(boolean[stat] increaseThisStat, boolean alwaysEquip)
+{
+	//want to equip best equipment that increases specified stat gains including out of combat
+	//should be frequently called by consume actions so try not to lose HP or MP, but will equip anyway if argument alwaysEquip is true
+	string maximizerStatement;
+	foreach st in increaseThisStat
+	{
+		string statWeight = "";
+		if(st == my_primestat())
+		{
+			if(my_level() < 13)
```

Comment:
check disreagardInstantKarma instead of level. If pref is set, we will want to get more primestat even at level 13

Reply:
guess so, more primestat than offstats or not might depend on which stat is primestat,
but anyway this would only make a difference when comparing 15% familiar scrapbook to fake washboard